proto/penumbra/penumbra/cnidarium/v1/cnidarium.proto

```diff
@@ -21,10 +21,19 @@ service QueryService {
 //
 // Proofs are only supported by key.
 message KeyValueRequest {
+  // The storage type to query.
+  enum StorageBackend {
+    STORAGE_BACKEND_UNSPECIFIED = 0;
+    STORAGE_BACKEND_JMT = 1;
+    STORAGE_BACKEND_NONVERIFIABLE = 2;
+  }
+
   // If set, the key to fetch from storage.
   string key = 2;
   // whether to return a proof
   bool proof = 3;
+  // Which storage to query.
+  StorageBackend storage_backend = 4;
```

Comment:
The idea of letting the user specify which storage layer they want to target is interesting, but I think the api would be simpler by just adding another message:

```protobuf
message NonVerifiableKeyValueRequest {
    message Key {
        bytes inner = 1;
    }

    Key key = 1;
}
```

This has a couple advantages:
1. We don't have to make callers `base64` key depending on which backend they target
2. We don't need an idle `proof` field that is only relevant for one backend
3. It makes our protobuf state query interface coherent with how we structured `StateRead`